astropy/units/quantity.py

```diff
@@ -42,6 +45,11 @@
 from .structured import StructuredUnit, _structured_unit_like_dtype
 from .utils import is_effectively_unity
 
+if TYPE_CHECKING:
+    from typing_extensions import Self
```

Comment:
Now that #15686 is merged this should be written as
```suggestion
    import sys
    
    if sys.version_info >= (3, 11):
        from typing import Self
    else:
        from typing_extensions import Self
```